ksp_plugin/renderer.cpp

```diff
@@ -74,52 +74,52 @@ Vessel const& Renderer::GetTargetVessel() const {
   return *target_->vessel;
 }
 
-not_null<std::unique_ptr<DiscreteTrajectory<World>>>
+DiscreteTraject0ry<World>
 Renderer::RenderBarycentricTrajectoryInWorld(
     Instant const& time,
-    DiscreteTrajectory<Barycentric>::Iterator const& begin,
-    DiscreteTrajectory<Barycentric>::Iterator const& end,
+    DiscreteTraject0ry<Barycentric>::iterator const& begin,
+    DiscreteTraject0ry<Barycentric>::iterator const& end,
     Position<World> const& sun_world_position,
     Rotation<Barycentric, AliceSun> const& planetarium_rotation) const {
   auto const trajectory_in_plotting_frame =
       RenderBarycentricTrajectoryInPlotting(begin, end);
   auto trajectory_in_world =
       RenderPlottingTrajectoryInWorld(time,
-                                      trajectory_in_plotting_frame->begin(),
-                                      trajectory_in_plotting_frame->end(),
+                                      trajectory_in_plotting_frame.begin(),
+                                      trajectory_in_plotting_frame.end(),
                                       sun_world_position,
                                       planetarium_rotation);
   return trajectory_in_world;
 }
 
-not_null<std::unique_ptr<DiscreteTrajectory<Navigation>>>
+DiscreteTraject0ry<Navigation>
 Renderer::RenderBarycentricTrajectoryInPlotting(
-    DiscreteTrajectory<Barycentric>::Iterator const& begin,
-    DiscreteTrajectory<Barycentric>::Iterator const& end) const {
-  auto trajectory = make_not_null_unique<DiscreteTrajectory<Navigation>>();
+    DiscreteTraject0ry<Barycentric>::iterator const& begin,
+    DiscreteTraject0ry<Barycentric>::iterator const& end) const {
+  DiscreteTraject0ry<Navigation> trajectory;
   for (auto it = begin; it != end; ++it) {
     auto const& [time, degrees_of_freedom] = *it;
     if (target_) {
-      auto const& prediction = target_->vessel->prediction();
-      if (time < prediction.t_min()) {
+      auto const prediction = target_->vessel->prediction();
```

Comment:
Why not `auto const& prediction = *target_->vessel->prediction();`?

Reply:
It really doesn't matter here, but I have tried to stay away from operating on segments directly, going through iterators instead.